src/sqlancer/duckdb/test/DuckDBCODDTestOracle.java

```diff
@@ -0,0 +1,1065 @@
+        // @SuppressWarnings("unchecked")
+        // DuckDBFunction<DuckDBAggregateFunction> subAggr = (DuckDBFunction<DuckDBAggregateFunction>) func;
+        // DuckDBExpression cr = subAggr.getArgs().get(0);
+        // if (cr instanceof DuckDBColumnReference) {
+        // columnType = ((DuckDBColumnReference) cr).getColumn().getType();
+        // }
+        // }
+        // }
+        // }
+        // }
+        // if (columnType != null) {
+        // idxTypeMap.put(index, columnType);
+        // }
+        // }
+        // }
+        return idxTypeMap;
+    }
+
+    private boolean compareResult(Map<String, List<DuckDBExpression>> r1, Map<String, List<DuckDBExpression>> r2) {
+        if (r1.size() != r2.size()) {
+            return false;
+        }
+        for (Map.Entry<String, List<DuckDBExpression>> entry : r1.entrySet()) {
+            String currentKey = entry.getKey();
+            if (!r2.containsKey(currentKey)) {
+                return false;
+            }
+            List<DuckDBExpression> v1 = entry.getValue();
+            List<DuckDBExpression> v2 = r2.get(currentKey);
+            if (v1.size() != v2.size()) {
+                return false;
+            }
+            List<String> v1Value = new ArrayList<>(
+                    v1.stream().map(c -> ((DuckDBConstant) c).toString()).collect(Collectors.toList()));
+            List<String> v2Value = new ArrayList<>(
+                    v2.stream().map(c -> ((DuckDBConstant) c).toString()).collect(Collectors.toList()));
+            Collections.sort(v1Value);
+            Collections.sort(v2Value);
+            // if (!v1Value.equals(v2Value)) {
+            // return false;
+            // }
+            for (int i = 0; i < v1Value.size(); ++i) {
+                if (!v1Value.get(i).equals(v2Value.get(i))) {
+                    String regx = "[+-]*\\d+\\.?\\d*[Ee]*[+-]*\\d+";
+                    Pattern pattern = Pattern.compile(regx);
+                    if (pattern.matcher(v1Value.get(i)).matches() && pattern.matcher(v2Value.get(i)).matches()) {
+                        if (!v1Value.get(i).substring(0, 6).equals(v2Value.get(i).substring(0, 6))) {
+                            return false;
+                        }
+                    } else {
+                        return false;
+                    }
+                }
+            }
+        }
+        return true;
+    }
+
+    public boolean useSubquery() {
+        if (this.state.getDbmsSpecificOptions().coddTestModel.isRandom()) {
+            return Randomly.getBoolean();
+        } else if (this.state.getDbmsSpecificOptions().coddTestModel.isExpression()) {
+            return false;
+        } else if (this.state.getDbmsSpecificOptions().coddTestModel.isSubquery()) {
+            return true;
+        } else {
+            System.out.printf("Wrong option of --coddtest-model, should be one of: RANDOM, EXPRESSION, SUBQUERY");
+            System.exit(1);
```

Comment:
I think using a switch-case on the enum would be clearer here, throwing an `AssertionError` in the default case.